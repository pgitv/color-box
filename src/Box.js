import React, { Component } from 'react';
import NewBoxForm from './NewBoxForm';
import './box.css';

class Box extends Component {
  constructor(props) {
    super(props);
    this.state = {
      width: this.props.width,
      height: this.props.height,
      backgroundColor: this.props.backgroundColor
    };
  }

  render() {
    let boxStyle = {
      backgroundColor: this.state.backgroundColor,
      width: this.state.width,
      height: this.state.height
    };
    return <div className="box" style={boxStyle} />;
  }
}

export default Box;
