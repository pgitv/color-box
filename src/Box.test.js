import React from 'react';
import ReactDOM from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Box from './Box';

// Smoke tests
it('renders without crashing', function() {
  shallow(<Box />);
});

// Snapshot tests
it('matches snapshot', function() {
  let wrapper = shallow(<Box />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
