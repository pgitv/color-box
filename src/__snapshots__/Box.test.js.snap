// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="box"
  style={
    Object {
      "backgroundColor": undefined,
      "height": undefined,
      "width": undefined,
    }
  }
/>
`;
